src/main/java/de/zalando/aruha/nakadi/repository/kafka/NakadiKafkaConsumer.java

```diff
@@ -52,25 +54,26 @@ public NakadiKafkaConsumer(final KafkaFactory factory, final String topic, final
         if (eventQueue.isEmpty()) {
             pollFromKafka();
         }
-
         return Optional.ofNullable(eventQueue.poll());
     }
 
     @Override
     public Map<String, String> fetchNextOffsets() {
-        return topicPartitions.stream().collect(Collectors.toMap(topicPartition ->
-                        Integer.toString(topicPartition.partition()),
-                    topicPartition -> Long.toString(kafkaConsumer.position(topicPartition))));
+        return topicPartitions
+                .stream()
+                .collect(Collectors.toMap(
+                        topicPartition -> Integer.toString(topicPartition.partition()),
+                        topicPartition -> Long.toString(kafkaConsumer.position(topicPartition))));
     }
 
     private void pollFromKafka() throws NakadiException {
         try {
             final ConsumerRecords<String, String> records = kafkaConsumer.poll(pollTimeout);
-            eventQueue = StreamSupport.stream(records.spliterator(), false).map(record ->
-                                              new ConsumedEvent(record.value(), record.topic(),
-                                                  Integer.toString(record.partition()),
-                                                  Long.toString(record.offset() + 1))).collect(Collectors.toCollection(
-                                              Lists::newLinkedList));
+            eventQueue = StreamSupport
+                    .stream(records.spliterator(), false)
+                    .map(record -> new ConsumedEvent(record.value(), record.topic(),
+                            Integer.toString(record.partition()), Long.toString(record.offset() + 1)))
```

Comment:
One could convert the `Cursor` into `KafkaCursor`. Or convert the values before the stream, I personally find it more readable.